tests/e2e/tests/pilot/ingressgateway_test.go

```diff
@@ -77,6 +77,42 @@ func TestGateway_HTTPIngress(t *testing.T) {
 	})
 }
 
+func TestGateway_HTTPSIngress(t *testing.T) {
+	if !tc.V1alpha3 {
+		t.Skipf("Skipping %s: v1alpha3=false", t.Name())
+	}
+
+	istioNamespace := tc.Kube.IstioSystemNamespace()
+	ingressGatewayServiceName := tc.Kube.IstioIngressGatewayService()
+
+	// Configure a route from us.bookinfo.com to "c-v2" only
+	cfgs := &deployableConfig{
+		Namespace: tc.Kube.Namespace,
+		YamlFiles: []string{
+			"testdata/v1alpha3/ingressgateway.yaml",
+			"testdata/v1alpha3/destination-rule-c.yaml",
+			"testdata/v1alpha3/rule-ingressgateway.yaml"},
+	}
+	if err := cfgs.Setup(); err != nil {
+		t.Fatal(err)
+	}
+	defer cfgs.Teardown()
+
+	runRetriableTest(t, "VersionRouting", defaultRetryBudget, func() error {
+		reqURL := fmt.Sprintf("https://%s.%s:443/c", ingressGatewayServiceName, istioNamespace)
```

Comment:
I pinged @ZackButcher and he mentioned that it should be sufficient to call uk.bookinfo.com and create a VS that routes `uk.bookinfo.com` to `istio-ingressgateway` .

Reply:
The question is how to call uk.bookinfo.com. It should be called in such a way that the sent SNI is "uk.bookinfo.com".

Reply:
One setup that should solve this problem is to create a VirtualService for the host `uk.bookinfo.com` that sends the traffic to the `istio-ingressgateway` service; then a second VirtualService bound to the gateway routes `uk.bookinfo.com` to `productpage.default.svc.cluster.local` (or whatever the real backend is).

The problem is that Gateways are intended to be transparent (e.g. in Ingress usecase client has no idea there's a Gateway or not) so the setup in our own tests looks a little funky.

Reply:
Does this mean we require a user to add this custom transport to their code? If so, is this something that we should classify as a usability bug?

Reply:
@cmluciano Sorry, I missed your question. In the real life, the users will have the _httbin.example.com_ hostname allocated to their cluster. The users will not have to use `--resolve` option. Since in our case, in the test environment, _httpbin.example.com_ is a fake domain name, we have to use `--resolve` option. So it is "a usability issue" in tests environments only, which is OK in my opinion.